src/components/accounts/RecoverAccountSeedPhrase.js

```diff
@@ -26,36 +25,48 @@ class RecoverAccountSeedPhrase extends Component {
 
     componentDidMount = () => {}
 
-    componentWillUnmount = () => {
-        this.props.clear()
-    }
-
     handleChange = (e, { name, value }) => {
         this.setState(() => ({
             [name]: value
         }))
+
+        this.props.clear()
     }
 
     handleSubmit = () => {
-
         if (!this.isLegit) {
             return false
         }
 
-        const accountId = this.state.accountId
-        this.props.recoverAccountSeedPhrase(this.state.seedPhrase, accountId)
+        this.props.recoverAccountSeedPhrase(this.state.seedPhrase)
             .then(({ error }) => {
                 if (error) return
                 this.props.refreshAccount()
                 this.props.redirectToApp()
             })
     }
 
+    checkInvalidSeedPhrase = () => this.props.requestStatus.errorMessage && this.props.requestStatus.errorMessage.includes('Cannot find matching public key')
+
+    handleRequestStatus = () => (
```

Comment:
nit: `handle` assumes method would actually have some side-effect. Seems like here it is more like just a getter for adjusted definition of request status. So we can be more explicit about it:

```js
get requestStatusWithFormValidation() {
    ...
}
```